projects/aca-content/src/lib/components/bulk-actions-dropdown/bulk-actions-dropdown.component.spec.ts

```diff
@@ -160,24 +167,27 @@ describe('BulkActionsDropdownComponent', () => {
         extensionService = TestBed.inject(AppExtensionService);
         spyOn(extensionService, 'getBulkActions').and.returnValue(of([mockItem]));
         fixture.detectChanges();
+        loader = TestbedHarnessEnvironment.loader(fixture);
       });
 
-      it('should run action on selection', () => {
-        spyOn(component, 'runAction').and.callThrough();
+      it('should run action on selection', async () => {
         spyOn(extensionService, 'runActionById');
-        const option = getElement(mockItem.id);
-        option.click();
+        const selectHarness = await loader.getHarness(MatSelectHarness);
+        await selectHarness.open();
+        await selectHarness.clickOptions(await selectHarness.getOptions()[0]);
+        await fixture.whenStable();
         fixture.detectChanges();
 
-        expect(component.runAction).toHaveBeenCalledWith(mockItem);
         expect(extensionService.runActionById).toHaveBeenCalledWith(mockItem.actions.click, {
           focusedElementOnCloseSelector: '.adf-context-menu-source'
         });
       });
 
-      it('should reset selection on bulkActionExecuted', () => {
-        const option = getElement(mockItem.id);
-        option.click();
+      it('should reset selection on bulkActionExecuted', async () => {
+        const selectHarness = await loader.getHarness(MatSelectHarness);
+        await selectHarness.open();
+        await selectHarness.clickOptions(await selectHarness.getOptions()[0]);
+        await fixture.whenStable();
```

Comment:
Since the selection code is getting reused, we can instead create a single method, selectOptionFromDropdown(selectionIndex: number), and then use that in both the test cases. That way, any change would need to be done in a single place, and we will have a utility method, that can be reused for any test cases that get added in the future

Reply:
Just pushed the changes. Great idea, thank you!